### PR TITLE
[Storage] Stress tests in pipeline?

### PR DESCRIPTION
**Pull request by kasobol-msft**
This PR adds basic coverage for APIs that are potentially memory-greedy, so they don't become greedier while code evolves.

Things to follow up separately observed during this experiment:
- java 11 requires much more memory for some reason (e.g. upload sync works with 500mb on java8 but needs 700mb on java11). Opened https://github.com/Azure/azure-sdk-for-java/issues/21631
- async scenarios run with insufficient heap just hang. See https://dev.azure.com/azure-sdk/internal/_build/results?buildId=899330&view=results . Created https://github.com/Azure/azure-sdk-for-java/issues/21629
- while experimenting with listing scenarios (not included here) I observed connection reset errors possibly not being re-tried by pageable types. Reopened https://github.com/Azure/azure-sdk-for-java/issues/21091

### CODE REVIEW

File: sdk/storage/tests.yml [hunk 1]
```diff
@@ -47,6 +50,19 @@ stages:
           Path: sdk/storage/platform-matrix.json
           Selection: sparse
           GenerateVMJobs: true
+      PostSteps:
+        - task: Maven@3
+          displayName: 'Run Stress Scenarios'
+          inputs:
+              mavenPomFile: sdk/storage/azure-storage-perf/pom.xml
+              options: -Dstorage-stress-scenarios
```
Comment (alzimmermsft) [inline sdk/storage/tests.yml:58]: I'd either add `$(DefaultOptions)` or `--no-transfer-progress`  to `options` to reduce the verbosity of this step.

File: sdk/storage/tests.yml [hunk 2]
```diff
@@ -47,6 +50,19 @@ stages:
           Path: sdk/storage/platform-matrix.json
           Selection: sparse
           GenerateVMJobs: true
+      PostSteps:
+        - task: Maven@3
```
Comment (alzimmermsft) [inline sdk/storage/tests.yml:54]: Do we want to run this task if the test run fails? Also, if this step fails do we want to fail the job?

Comment (kasobol-msft) [reply sdk/storage/tests.yml:54]: I think we can live with this not running if tests fail. We want to have green build. I'll check the later.

**Review by mikeharder — APPROVED** [review]
If this works as a reliable test, it looks good to me.

File: sdk/storage/tests.yml (hunk 2 above)
Comment (kasobol-msft) [reply sdk/storage/tests.yml:54]: image

File: sdk/storage/azure-storage-perf/memory-stress-scenarios.ps1 [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    Write-Host "Executing '$Scenario' with '$HeapSize' heap"
+    Invoke-Expression "& '$JavaPath' -Xms$HeapSize -Xmx$HeapSize -jar '$PerfJarPath' $Scenario"
+    if ($LASTEXITCODE -ne 0) {
+        Write-Host "Scenario failed, exiting"
+        exit 1
+    }
+}
+
+Run-Scenario "700m" "uploadoutputstream --warmup 0 --duration 1 --size 1048576000 --sync"
+Run-Scenario "700m" "uploadblob --warmup 0 --duration 1 --size 1048576000"
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-perf/memory-stress-scenarios.ps1:23]: for some reason going below 700m doesn't work on Java 11. Java 8 is less greedy...

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java [hunk 4]
```diff
@@ -249,7 +249,7 @@ private static void runLoop(PerfStressTest<?> test, int index, long endNanoTime)
                 completedOperations[index]++;
                 lastCompletionNanoTimes[index] = System.nanoTime() - startNanoTime;
             })
-            .take(Duration.ofNanos(endNanoTime - startNanoTime))
+            .takeWhile(i -> System.nanoTime() < endNanoTime || completedOperations[index] == 0)
```
Comment (kasobol-msft) [inline common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:252]: this makes async loop to complete at least 1 operation.

**Review by gapra-msft — APPROVED** [review]

**Review by mikeharder — APPROVED** [review]

**Pull request merged**